sys/include/net/netstats/neighbor.h

```diff
@@ -0,0 +1,220 @@
+/*
+ * Copyright (C) 2017 Koen Zandberg <koen@bergzand.net>
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser General
+ * Public License v2.1. See the file LICENSE in the top level directory for
+ * more details.
+ */
+
+/**
+ * @defgroup    net_gnrc_netdev_stats  Netdev stats group
+ * @ingroup     net_gnrc_netdev
+ * @brief       Records statistics about link layer neighbors
+ * @{
+ *
+ * @file
+ * @brief       Neighbor stats definitions
+ *
+ * @author      Koen Zandberg <koen@bergzand.net>
+ */
+#ifndef NET_NETSTATS_NEIGHBOR_H
+#define NET_NETSTATS_NEIGHBOR_H
+
+#include <string.h>
+#include "net/netdev.h"
+#include "xtimer.h"
+#include "timex.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief Result of the transmission
+ * @{
+ */
+typedef enum {
+    NETSTATS_NB_BUSY,       /**< Failed due to medium busy */
+    NETSTATS_NB_NOACK,      /**< Failed due to no ack received */
+    NETSTATS_NB_SUCCESS,    /**< Succesful transmission */
+} netstats_nb_result_t;
+/** @} */
+
+/**
+ * @brief Max length of a L2 address
+ */
+#define NETSTATS_NB_L2_ADDR_MAX_SIZE      (8)
+
+/**
+ * @name @ref EWMA parameters
+ * @{
+ */
+#define NETSTATS_NB_EWMA_SCALE            100   /**< Multiplication factor of the EWMA */
```

Comment:
Not sure if there's value to make all these defines configurable on compile time ..
Wouldn't hurt to wrap them around `#ifndef ..#endif`?

Reply:
Most of these make sense to have configurable. All of them control the timings in some way. For a mostly static network, timings can probably be a bit longer and for more dynamic you probably want some shorter timings and a more responsive averaging, maybe even a higher noack penalty.

TL;DR: Yes.